web/src/lib/components/assets/thumbnail/image-thumbnail.svelte

```diff
@@ -11,48 +11,38 @@
 	export let curve = false;
 	export let shadow = false;
 	export let circle = false;
-	let loading = true;
 
-	let imageElement: HTMLImageElement;
+	let complete = false;
 
-	onMount(() => {
-		if (thumbhash) {
-			lazyLoad(imageElement, {
-				hash: thumbhash,
-				hashType: 'thumbhash'
-			});
-		}
-	});
+	const decodeBase64 = (s: string) => Uint8Array.from(atob(s), (c) => c.charCodeAt(0));
 </script>
 
-{#if thumbhash}
-	<img
-		style:width={widthStyle}
-		style:height={heightStyle}
-		data-src={url}
-		alt={altText}
-		class="object-cover"
-		class:rounded-lg={curve}
-		class:shadow-lg={shadow}
-		class:rounded-full={circle}
-		draggable="false"
-		bind:this={imageElement}
-	/>
+<img
+	style:width={widthStyle}
+	style:height={heightStyle}
+	src={url}
+	alt={altText}
+	class="absolute object-cover transition-opacity duration-300"
+	class:rounded-lg={curve}
+	class:shadow-lg={shadow}
+	class:rounded-full={circle}
+	class:opacity-0={!thumbhash && !complete}
+	draggable="false"
+	use:imageLoad
+	on:image-load|once={() => (complete = true)}
+/>
 
-	<!-- not everthing yet has thumbhash support so the old method is kept -->
-{:else}
+{#if thumbhash && !complete}
```

Comment:
Why the second img element? Can't the thumbnail be loaded in onMount and replace `src`?

Reply:
I don't think the transition works when it is the same element

Reply:
It's not possible to transition images with just a single image element. There is no control over how the underlying API handles changes to the `src` attribute.

Reply:
FWIW though, I do agree with you. It would be nice if svelte could just swap them dynamically or something. I initially started with something like:

```ts
const img = new Image();
img.src = url;
```

and

```svelte
{#await img.decode()}
<img thumbhash>
{:then}
<img src:{url}>
{/await}
```

In practise, it's just not very effective. It loads the target image multiple times and can get a little messy. I think this implementation is ultimately quite nice. It allows really easy compatibility for assets which don't have thumbhashes, using the old behaviour, it's quite simple and doesn't end up loading the target image twice. It also will eventually remove the thumbhash from the DOM entirely, and so the result is just one image anyway.